pkg/services/libraryelements/database.go

```diff
@@ -434,12 +444,28 @@ func (l *LibraryElementService) patchLibraryElement(c *models.ReqContext, cmd pa
 		if elementInDB.Version != cmd.Version {
 			return errLibraryElementVersionMismatch
 		}
+		updateUID := cmd.UID
+		if len(updateUID) == 0 {
+			updateUID = uid // if UID isn't provided in the PATCH request let's use the existing UID
+		}
+		if !util.IsValidShortUID(updateUID) {
+			return errLibraryElementInvalidUID
+		} else if util.IsShortUIDTooLong(updateUID) {
+			return errLibraryElementUIDTooLong
+		}
+		if updateUID != uid {
```

Comment:
I am thinking we can move this directly upper 
```
	if len(updateUID) == 0 {
      // if UID isn't provided in the PATCH request let's use the existing UID
			updateUID = uid 
	} else {
     // if UID is provided in the PATCH we need to check whether it exists already.
			_, err := getLibraryElement(l.SQLStore.Dialect, session, updateUID, c.SignedInUser.OrgId)
			if !errors.Is(err, errLibraryElementNotFound) {
				return errLibraryElementAlreadyExists
			}
// other checks
 }
```
might be a tad more direct forward and one less condition.  LGTM otherwise :)

Reply:
Changed the code so we don't check existing UIDs.